nixos/modules/services/hardware/xow.nix

```diff
@@ -0,0 +1,46 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.services.xow;
+in {
+  options.services.xow = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Whether to enable xow or not.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users.uinput = {
+      group = "uinput";
+      isSystemUser = true;
+    };
+
+    users.groups.uinput = {};
+
+    services.udev.packages = lib.singleton (pkgs.writeTextFile {
+      name = "xow-uinput-rules";
+      destination = "/etc/udev/rules.d/99-uinput.rules";
+      text = ''
+        SUBSYSTEM=="misc", KERNEL=="uinput", MODE="0660", GROUP="uinput", OPTIONS+="static_node=uinput"
```

Comment:
Since this is not specific to `xow` I think a separate module is a good idea. So how about a module in `services/hardware/uinput.nix` adding an option `hardware.uinput` which adds the `uinput` group, sets this udev rule with `services.udev.extraRules` (simpler than with `*.packages`), and sets `boot.kernelModules = [ "uinput" ]` (apparently this is needed sometimes, see the `steam-hardware.nix` module). Then both the `xow` module and the `steam-hardware.nix` one can enable this option. And then finally, this module can create a user with name `xow` being part of the `uinput` group, to be used for running this service.

Reply:
Related: https://github.com/NixOS/nixpkgs/issues/70471, ping @cdepillabout @nyanloutre @illegalprime

Reply:
Okay, I did exactly what you said, but I didn't adapt the steam part yet. It seems to work fine and feels more general than my hacky uinput solution. :)